impeller/geometry/geometry_benchmarks.cc

```diff
@@ -6,381 +6,308 @@
-                        {339.115, 114.151})
-          .CubicCurveTo({338.251, 114.379}, {337.34, 114.516},
-                        {336.448, 114.516})
-          .CubicCurveTo({335.761, 114.516}, {335.072, 114.527},
-                        {334.384, 114.513})
-          .CubicCurveTo({334.125, 114.508}, {333.862, 114.462},
-                        {333.605, 114.424})
-          .CubicCurveTo({332.865, 114.318}, {332.096, 114.184},
-                        {331.41, 113.883})
-          .CubicCurveTo({330.979, 113.695}, {330.442, 113.34},
-                        {330.672, 112.813})
-          .CubicCurveTo({331.135, 111.755}, {333.219, 112.946},
-                        {334.526, 113.833})
-          .CubicCurveTo({334.54, 113.816}, {334.554, 113.8}, {334.569, 113.784})
-          .CubicCurveTo({333.38, 112.708}, {331.749, 110.985},
-                        {332.76, 110.402})
-          .CubicCurveTo({333.769, 109.82}, {334.713, 111.93},
-                        {335.228, 113.395})
-          .CubicCurveTo({334.915, 111.889}, {334.59, 109.636},
-                        {335.661, 109.592})
-          .CubicCurveTo({336.733, 109.636}, {336.408, 111.889},
-                        {336.07, 113.389})
-          .CubicCurveTo({336.609, 111.93}, {337.553, 109.82},
-                        {338.563, 110.402})
-          .CubicCurveTo({339.574, 110.984}, {337.942, 112.708},
-                        {336.753, 113.784})
-          .CubicCurveTo({336.768, 113.8}, {336.782, 113.816},
-                        {336.796, 113.833})
-          .CubicCurveTo({338.104, 112.946}, {340.187, 111.755},
-                        {340.65, 112.813})
-          .CubicCurveTo({340.71, 112.95}, {340.728, 113.102},
-                        {340.678, 113.245})
-          .Close()
-          .MoveTo({346.357, 106.771})
-          .CubicCurveTo({346.295, 104.987}, {347.924, 104.139},
-                        {347.924, 104.139})
-          .CubicCurveTo({347.924, 104.139}, {347.982, 105.931},
-                        {346.357, 106.771})
-          .Close()
-          .MoveTo({347.56, 106.771})
-          .CubicCurveTo({347.498, 104.987}, {349.127, 104.139},
-                        {349.127, 104.139})
-          .CubicCurveTo({349.127, 104.139}, {349.185, 105.931},
-                        {347.56, 106.771})
-          .Close()
-          .TakePath();
+namespace {
+/// A path with many connected cubic components, including
+/// overlaps/self-intersections/multi-contour.
+Path CreateCubic();
+/// Similar to the path above, but with all cubics replaced by quadratics.
+Path CreateQuadratic();
+}  // namespace
+
+template <class... Args>
+static void BM_Polyline(benchmark::State& state, Args&&... args) {
+  auto args_tuple = std::make_tuple(std::move(args)...);
+  auto path = std::get<Path>(args_tuple);
+  bool tessellate = std::get<bool>(args_tuple);
+
   size_t point_count = 0u;
   size_t single_point_count = 0u;
+  Tessellator tess;
```

Comment:
If you create the tessellator in the benchmark you're going to mostly be measuring the cost of bucket allocation. Since we cache this across frames and across tesselations I don't think it will be representative

Reply:
I'm only doing it once, but I could just do it statically.

Reply:
Moved this to a static var.

Reply:
(FWIW, it did make a small difference in the benchmark on my M1, but not much - but it's fair that we exclude that for the purpose of these benchmarks)